app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt

```diff
@@ -42,11 +41,23 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
         ActivityResultContracts.RequestMultiplePermissions()
     ) { isGranted: Map<String, Boolean> ->
         if (isGranted.all { it.value }) {
-            binding.swiperefresh.isRefreshing = true
-            setDefaultSubscriptionId()
-            onRefresh()
+            // phone permissions granted
+            if(isGranted.containsKey("android.permission.CALL_PHONE") || isGranted.containsKey("android.permission.READ_PHONE_STATE")){
```

Comment:
You can use a constant instead of a string here and below: `isGranted.containsKey(android.Manifest.permission.CALL_PHONE)`